### ISSUE TITLE
15.4.1 breaks URL parameters with duplicate names

### ISSUE THREAD

**Issue by oyvindmidtbo** [netlify/cli#5790]
### Describe the bug

After updating from 15.4.0 to 15.4.1, it's no longer possible to use multiple URL parameters with the same name.

### Steps to reproduce

```
import { APIGatewayProxyEvent, Handler } from 'aws-lambda';

const handler: Handler = async (event: APIGatewayProxyEvent) => {
  console.log(event.queryStringParameters);

  return {
    statusCode: 200,
    body: JSON.stringify({ message: 'Hello World' }),
  };
};

export { handler };
```

1. Install netlify-cli 15.4.0 and run a function, for example `localhost:8888/nf-functions/myfunction?param1=foo&param1=bar`
2. In the handler, I console.log `event.queryStringParameters`: `{ param1: 'foo, bar' }`
3. Install netlify-cli 15.4.1 and run the same function with the same parameters.
4. The new console.log: `{ param1: 'foo' }`

### Environment

System:
    OS: macOS 13.4
    CPU: (12) arm64 Apple M2 Max
    Memory: 3.57 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
  npmPackages:
    netlify-cli: ^15.4.1 => 15.4.1

**Comment by hemm1** [comment]
I work on the same team and just to avoid initial confusion: 

This behavior only appears when we are accessing the function through our own redirect. We have the following redirect in the `netlify.toml` file:

```
[[redirects]]
  from = "/nf-functions/*"
  to = "/.netlify/functions/:splat"
  status = 200
```

When we access the function without the redirect (`localhost:8888/.netlify/functions/myfunction?param1=foo&param1=bar`), we get access to the parameters in the same way as before.

Also, as you might expect, this does not affect our functions in production in any way. They still work as before.

**Comment by eduardoboucas** [comment]
Thanks for the detailed report. This should be fixed in 15.5.1 — could you give it a try?

If you're still seeing the issue, please let me know and I'll reopen the ticket.
